csrc/index_compute.cpp

```diff
@@ -2138,6 +2138,77 @@ Val* Index::getProducerStridedIndices(
   }
 }
 
+namespace {
+
+bool shouldUseTensorIndexer(
```

Comment:
This is getting rather convoluted, mostly because of the lack of proper support of ldmatrix and stmatrix.